### PR TITLE
Jacqueschen1/team leads fields

### LINKED ISSUE #238: Add Team Leads Field in Team Model
Need a field to keep track of team leads in a team

### PR DESCRIPTION
**Pull request by jacqueschen1**
# Pull Request

## Description

Added team leads field in team model, started on team commands 

## Testing

**If testing this change requires extra setup, please document it here:**

## Ticket(s)
Closes #238 

(Create a copy of that line for each Github Issue affected,
and replace "Affects" with "Closes" if merging this will close the relevant ticket.)

### CODE REVIEW

**Comment by rwblickhan** [comment]
Looks like there's a Markdown linting error.

File: command/commands/team.py [hunk 1]
```diff
@@ -163,3 +172,35 @@ def handle(self, command, user_id):
...
+                msg += "platform: {}, ".format(param_list["platform"])
+                team.platform = param_list["platform"]
+            if param_list["channel"] is not None:
+                msg += "add channel"
+                # stub: cannot finish until teamMember PR is pushed
+            self.facade.store_team(team)
+            return msg, 200
+        except GithubAPIException as e:
+            logging.error("team created unsuccessfully")
+            return "Team created unsuccessfully with the following error"\
```
Comment (rmcreyes) [inline command/commands/team.py]: Shouldn't we expect a tuple here?

**Comment by jacqueschen1** [comment]
Just realised that the Piplock file got changed, not sure why though

File: command/commands/team.py [hunk 2]
```diff
@@ -163,3 +172,35 @@ def handle(self, command, user_id):
...
+            if param_list["platform"] is not None:
+                msg += "platform: {}, ".format(param_list["platform"])
+                team.platform = param_list["platform"]
+            if param_list["channel"] is not None:
+                msg += "add channel"
+                # stub: cannot finish until teamMember PR is pushed
+            self.facade.store_team(team)
+            return msg, 200
+        except GithubAPIException as e:
+            logging.error("team created unsuccessfully")
```
Comment (rwblickhan) [inline command/commands/team.py]: Maybe `"team creation unsuccessful"`? " "Team created unsuccessfully" sounds strange to my ears. Same with the later instances.

File: docs/TeamCommands.md [hunk 3]
```diff
@@ -84,19 +83,26 @@ easy slack autocomplete.
...
-> **Note:** This command does not remove a member of the team from Github.
-
 Remove a user from a team. Team Leads can only remove users from teams that they
 are a part of, but admins can remove users from any team. `SLACK_ID` is the
 `@`-name, for easy slack autocomplete.
 
+### Lead (Team Lead\* and Admin only)
+
+```sh
+/rocket team lead SLACK_ID
```
Comment (rwblickhan) [inline docs/TeamCommands.md]: Which team is this applied to?

Comment (jacqueschen1) [reply docs/TeamCommands.md]: Oh my bad, didnt add that field

File: docs/TeamCommands.md [hunk 4]
```diff
@@ -84,19 +83,28 @@ easy slack autocomplete.
...
-Remove a user from a team. Team Leads can only remove users from teams that they
+Remove a user from a team, removes them as Team Lead if they were one.
+Team Leads can only remove users from teams that they
 are a part of, but admins can remove users from any team. `SLACK_ID` is the
 `@`-name, for easy slack autocomplete.
 
+### Lead (Team Lead\* and Admin only)
+
+```sh
+/rocket team lead GITHUB_TEAM_NAME SLACK_ID [--remove]
```
Comment (rwblickhan) [inline docs/TeamCommands.md]: Perhaps there should be an additional command to check the lead of a given team? I suppose that's listed in `view` though.

Comment (jacqueschen1) [reply docs/TeamCommands.md]: Yah I guess it would be shown in view

**Pull request merged**